### PR TITLE
Have you ever seen an awesome Cheep That you wanted to Answer directly? Introducing Comments!

### PR DESCRIPTION
**Pull request by DragonFlyersx**
# Have you ever seen an awesome Cheep That you wanted to Answer directly? Introducing Comments!

With Comments you now have the ability to comment on the every cheep

This means that you can keep a conversation going about a specific topic without blowing up your timeline

The Comments will be located with " View Comments" These will displayed underneath every single cheep on the site. 

When pressing view comments you will be taken to a new page that handles and shows all the comments 

Skærmbillede 2024-11-30 140150

Here you will be able to post your new comments and add them to the thread

If you want to remove a message you are able to do so by clicking delete

### For the people Behind the scenes
- Added Comments
- Added Comment pages
- Added a redirect from our cheeps to their comment pages
- Added the Ability to delete Comments
- Added Comments to be displayed from when they where posted
- Added Comment.cs and CommentDTO
- Added 3 E2E test to test functionality of the comments
- Added the ability to post a comment on a specific Cheep
-  Added the ability for Comments do be downloaded with userdata

### CODE REVIEW

**Review by Nikolaj787 — APPROVED** [review]
Great new feature and also great job adding it so that the user can also download their comments as part of one's personal data. And also good job with the merge, since we have added a lot of new stuff this week it could easily go wrong :D

File: src/Chirp.Web/Areas/Identity/Pages/Account/Manage/DownloadPersonalData.cshtml.cs [hunk 1]
```diff
@@ -100,6 +100,14 @@ public async Task<IActionResult> OnPostAsync()
                 j++;
             }
 
+            var k = 0;
+            var ListOfComments = await _service.RetrieveAllCommentsFromAnAuthor(user.UserName);
+            foreach (var Comment in ListOfComments)
+            {
+                csvContent.AppendLine($"Comment {k}: Comment.Text {Comment.Text}, Author: {Comment.Author.Name}, Comment TimeStamp: {Comment.FormattedTimeStamp}");
```
Comment (niko391a) [inline src/Chirp.Web/Areas/Identity/Pages/Account/Manage/DownloadPersonalData.cshtml.cs:107]: Great that you guys also added comments as part of the downloadable data :)

**Review by niko391a — APPROVED** [review]
Nice implementation in genreal, it was a good idea to have a separate page for a comment thread.

**Review by RasmusAChr — CHANGES_REQUESTED** [review]
Great implementation with comments! 

I think the decision with creating a dedicated page is much better than if the comments where nested underneath the Cheep.

However if im not wrong (looking from phone), the comment page doesnt show if there is an image in the cheep. You can talk to @Nikolaj787 about how to implement that or me in the evening tomorrow. You can also just look at the implementation in public timeline.

Furthermore I kind of miss a indicator whether or not a Cheep has any comments. Maybe it can be done like "Comments (7)". Maybe its just me, but it could be a QoL improvement, and i dont think its a hard implementation. (You can just use the same query for fetching comment and add a `.Count` to get the amount of comments.

Otherwise im really impressed, and now stop disliking my comments #StopMobning xD

**Comment by DragonFlyersx** [comment]
> Great implementation with comments! 
> 
> I think the decision with creating a dedicated page is much better than if the comments where nested underneath the Cheep.
> 
> However if im not wrong (looking from phone), the comment page doesnt show if there is an image in the cheep. You can talk to @Nikolaj787 about how to implement that or me in the evening tomorrow. You can also just look at the implementation in public timeline.
> 
> Furthermore I kind of miss a indicator whether or not a Cheep has any comments. Maybe it can be done like "Comments (7)". Maybe its just me, but it could be a QoL improvement, and i dont think its a hard implementation. (You can just use the same query for fetching comment and add a `.Count` to get the amount of comments.
> 
> Otherwise im really impressed, and now stop disliking my comments #StopMobning xD



I will try to look into it and see if it possible to add Images might be cool to add aswell I will have a look but I don't want to create a new migration since that might be nessecary

**Review by RasmusAChr — APPROVED** [review]
tenor

**Pull request merged**